oio/xcute/orchestrator.py

```diff
@@ -80,22 +93,23 @@ def run_forever(self):
 
         self.threads[beanstalkd_thread.ident] = beanstalkd_thread
 
-        # restart running jobs
-        self.logger.debug('Look for unfinished jobs')
-        orchestrator_jobs = \
-            self.backend.list_orchestrator_jobs(self.orchestrator_id)
-
-        for job_info in orchestrator_jobs:
-            job_id = job_info['job']['id']
-            self.logger.info('Found running job %s', job_id)
-            self.handle_running_job(job_info)
-
         # start processing replies
         listen_thread = threading.Thread(target=self.listen)
         listen_thread.start()
-
         self.threads[listen_thread.ident] = listen_thread
 
+        # restart running jobs
+        orchestrator_jobs = None
+        if self.running:
```

Comment:
Early return
There is no point putting everything inside ifs

Reply:
If we return here, we don't wait for `beanstalkd_thread` and `listen_thread` to stop. To exit properly, we have to wait for all started threads to stop.

Reply:
@AymericDu You can put this code inside a function and wait for the threads to stop outside the function?